Dynatrace/dynatrace-operator
# Refactor node controllers

## Description

Please include the following:

This PR implements some housekeeping work around node controller, including: 
- remove unused code 
- remove nodeDeletionPredicate from the event filter and handle this case properly in the Reconcile loop
- add more tests
- migrate from `time.Now` to `timeprovider`. 



## How can this be tested?

1) Deploy the operator and install dynakube. Then, delete the node and make sure we send a mark for the termination request.
2) Drain/Cordon node and wait ~1h and check logs.

## Checklist
